tests/integration-tests/tests/budgets/test_budgets.py

```diff
@@ -0,0 +1,141 @@
+import boto3
+import pytest
+from assertpy import assert_that
+
+
+@pytest.mark.usefixtures("os", "instance", "scheduler")
+def test_update_add_and_remove_budget(pcluster_config_reader, clusters_factory, test_datadir, region):
+    # Create the initial cluster without a budget definition
+    init_config_file = pcluster_config_reader(config_file="no_budgets.yaml")
+    cluster = clusters_factory(init_config_file)
+
+    # Update the cluster to add a budget
+    new_config_file = pcluster_config_reader(config_file="budget_with_notifications.yaml")
+    cluster.update(str(new_config_file), force_update="true")
+
+    # Get the account id to check for the created budget
+    sts_client = boto3.client("sts")
+    account_id = sts_client.get_caller_identity().get("Account")
+
+    # Try to describe the budget created by the cluster
+    budget_name_beginning = f"{cluster.name}-0-{region}"
+    budget_client = boto3.client("budgets")
+    budgets_description = budget_client.describe_budgets(AccountId=account_id)["Budgets"]
+    assert_that(
+        any([budget["BudgetName"].startswith(budget_name_beginning) for budget in budgets_description])
+    ).is_true()
+
+    # Update the cluster again with the initial file, removing the budget
+    cluster.update(str(init_config_file), force_update="true")
+
+    # Try to describe the budget again (should fail this time)
+    budgets_description = budget_client.describe_budgets(AccountId=account_id)["Budgets"]
+    assert_that(
+        any([budget["BudgetName"].startswith(budget_name_beginning) for budget in budgets_description])
+    ).is_false()
+
+
+@pytest.mark.usefixtures("os", "instance", "scheduler")
+def test_update_budget_properties(pcluster_config_reader, clusters_factory, test_datadir, region):
+    # Create initial cluster with 3 budgets (one of each BudgetCategory)
+    init_config_file = pcluster_config_reader(config_file="all_budget_categories.yaml")
+    cluster = clusters_factory(init_config_file)
+
+    # Update the cluster to change the budget properties
+    new_config_file = pcluster_config_reader(config_file="all_budget_categories_updated.yaml")
+    cluster.update(str(new_config_file), force_update="true")
+
+    # The budgets from the updated configuration
+    budget_list = cluster.config["DevSettings"]["Budgets"]
+    custom_budget = budget_list[0]
+    cluster_budget = budget_list[1]
+    queue_budget = budget_list[2]
```

Comment:
nit: the code can be combined into one line: 
custom_budget, cluster_budget, queue_budget = budget_list[0], budget_list[1], budget_list[2]